### PR TITLE
Fix changelog divergence for 2.x on main

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- Fixes changelog divergence for `2.x` on `main`

### Related Issues
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by saratvemulapalli — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -86,13 +79,23 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased 2.x]
```
Comment (saratvemulapalli) [inline CHANGELOG.md:80]: Nit: Update `[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2.5...2.x` to `[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2.7...2.x`

Comment (kotwanikunal) [reply CHANGELOG.md:80]: Good catch!

**Review by mch2 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -86,13 +79,23 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 ### Added
-- [Extensions] Moving Extensions APIs to protobuf serialization. ([#6960](https://github.com/opensearch-project/OpenSearch/pull/6960))
-- [Extensions] Add IdentityPlugin into core to support Extension identities ([#7246](https://github.com/opensearch-project/OpenSearch/pull/7246))
+- [Extensions] Moving Extensions APIs to support cross versions via protobuf. ([#6960](https://github.com/opensearch-project/OpenSearch/pull/6960))
```
Comment (saratvemulapalli) [inline CHANGELOG.md:82]: My apologies, I've just merged in another PR #7403 which fixes this. 
You might end up in merge conflicts.

File: CHANGELOG.md [hunk 3]
```diff
@@ -106,4 +109,4 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Security
 
 [Unreleased 3.0]: https://github.com/opensearch-project/OpenSearch/compare/2.x...HEAD
-[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2.5...2.x
+[Unreleased 2.x]: https://github.com/opensearch-project/OpenSearch/compare/2....2.x
```
Comment (saratvemulapalli) [inline CHANGELOG.md:112]: Lost `7` in `2.7` :/

Comment (kotwanikunal) [reply CHANGELOG.md]: Sorry. Got drowned in too many branches. Updated

**Pull request merged**